flycheck.el

```diff
@@ -6920,6 +6920,49 @@ See URL `http://www.erlang.org/'."
    (error line-start (file-name) ":" line ": " (message) line-end))
   :modes erlang-mode)
 
+(defun locate-rebar3-project-root (file-name)
+  "Return directory where the root =rebar.config= for the source FILE-NAME is located."
+  (let* ((possibly-nested-project-dir
+          (locate-dominating-file file-name "rebar.config"))
+         (possible-parent-dir
+          (if possibly-nested-project-dir
+              (locate-dominating-file
+               (expand-file-name ".." possibly-nested-project-dir)
+               "rebar.config")
+            nil)
```

Comment:
I've been thinking about whether we could simplify this code, to only call `locate-dominating-file` once.  Would that be possible?